components/core/src/clp_s/search/Output.cpp

```diff
@@ -580,83 +576,142 @@ bool Output::evaluate_var_string_filter(
+    switch (item.type()) {
+        case ondemand::json_type::object: {
+            ondemand::object nested_object = item.get_object();
+            if (evaluate_array_filter(nested_object, op, unresolved_tokens, cur_idx, operand)) {
+                match = true;
+            }
+        } break;
+        case ondemand::json_type::array: {
+            ondemand::array nested_array = item.get_array();
+            if (evaluate_array_filter(nested_array, op, unresolved_tokens, cur_idx, operand)) {
+                match = true;
+            }
+        } break;
+        case ondemand::json_type::string: {
+            if (true == m_maybe_string && unresolved_tokens.size() == cur_idx
+                && wildcard_match(item.get_string().value(), m_array_search_string))
             {
-                match |= evaluate_array_filter(
-                        value,
-                        op,
-                        unresolved_tokens,
-                        cur_idx + 1,
-                        operand,
-                        false
-                );
-            } else if (array_or_object) {
-                match |= evaluate_array_filter(
-                        value,
-                        op,
-                        unresolved_tokens,
-                        cur_idx,
-                        operand,
-                        false
-                );
+                match = op == FilterOperation::EQ;
             }
-        } else if (((array_or_object && cur_idx == unresolved_tokens.size())
-                    || (!array_or_object && cur_idx == unresolved_tokens.size() - 1
-                        && i.key() == unresolved_tokens[cur_idx].get_token())))
-        {
-            std::string tmp_string;
-            int64_t tmp_int;
-            double tmp_float;
-            bool tmp_bool;
-            if (FilterOperation::EXISTS == op || FilterOperation::NEXISTS == op
-                || (value.is_number_integer() && operand->as_int(tmp_int, op)
-                    && eval(op, value.get<int64_t>(), tmp_int))
-                || (value.is_number_float() && operand->as_float(tmp_float, op)
-                    && eval(op, value.get<double>(), tmp_float))
-                || (value.is_boolean() && operand->as_bool(tmp_bool, op)
-                    && eval(op, value.get<bool>(), tmp_bool)))
+        } break;
+        case ondemand::json_type::number: {
+            if (false == m_maybe_number || unresolved_tokens.size() != cur_idx) {
+                break;
+            }
+            ondemand::number number = item.get_number();
+            if (number.is_double()) {
+                double tmp_double;
+                operand->as_float(tmp_double, op);
+                match = eval(op, number.get_double(), tmp_double);
+            } else if (number.is_uint64()) {
+                int64_t tmp_int;
+                operand->as_int(tmp_int, op);
+                match = eval(op, number.get_uint64(), tmp_int);
```

Comment:
Do you think I should bit cast the operand to a uint64 here?  The "correct" solution is probably to go through the code base and properly support unsigned integers at the storage and AST levels (unsigned int node type, and as_unsigned method on Literal or something), but I'm not sure how much of that we want to pull into this PR.

Reply:
For now I'm going to do a bit cast and add a TODO

Reply:
I think we can make an assumption that only INT64_MIN to INT64_MAX are supported? Because in simdjson the number can be uint64_t only if it's larger than INT64_MAX, but we then cast it to int64_t which results in a negative value.